packages/flutter_test/lib/src/_leak_tracking.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:leak_tracker/leak_tracker.dart';
+
+import 'leak_tracking.dart';
+import 'test_async_utils.dart';
+import 'widget_tester.dart';
+
+/// Unit test friendly wrapper for [withLeakTracking] with Flutter specific functionality.
+///
+/// The method will fail with test matcher error if wrapped code contains memory leaks.
+///
+/// See details in documentation for `withLeakTracking` at
+/// https://github.com/dart-lang/leak_tracker/blob/main/lib/src/orchestration.dart#withLeakTracking
+///
+/// The Flutter related enhancements are:
+/// 1. Listens to [MemoryAllocations] events.
+/// 2. Uses `asyncCodeRunner` for async call for leak detection.
+Future<void> withFlutterLeakTracking(
+  DartAsyncCallback callback, {
+  required WidgetTester tester,
+  LeakTrackingFlutterTestConfig? config,
+}) async {
+  // Leak tracker does not work for web platform.
+  if (kIsWeb) {
+    await callback();
```

Comment:
should we disallow using the leak tracker on the web then and throw? This may give people the false sense that they don't have leaks if the only test on the web. Also, this limitation needs to be mentioned in the docs.

Reply:
Alternatively, print a warning the first time leak tracking is requested in a web target

Reply:
The same tests are exercised on many platforms. It is more convenient  for users to just pass leak tracking if the platform is web.

Printed warning.

Reply:
That warning doesn't appear to be actionable: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#only-log-actionable-messages-to-the-console